File: MCL_by_python/CommonFunctions.py
import math

a = 8.0 * (math.pi - 3) / (3 * math.pi * (4 - math.pi))
s_SquareRoot2 = math.sqrt(2.0)


def Sign(x):
    if x > 0:
        return 1
    elif x < 0:
        return -1
    else:
        return 0


def ErrorFunction(x):
    xSquared = x * x

    er = Sign(x) * math.sqrt(1 - math.exp(-xSquared * (4.0 / math.pi + a * xSquared) / (1 + a * xSquared)))

    return er


def DistributionFunction(x, mean, variance):
    return 0.5 * (1 + ErrorFunction((x - mean) * 1.0 / (math.sqrt(variance) * s_SquareRoot2)))

File: MCL_by_Cpp/MCL_C++/CommonFunctions.cpp
#include "stdafx.h"
#include <iostream>
#include <cmath>
#include "CommonFunctions.h"
#include "Points.h"

double a = 8.0 * (p_PI - 3) / (3 * p_PI * (4 - p_PI));
double s_SquareRoot2 = sqrt(2.0);


int Sign(double x)
{
    if (x > 0)
        return 1;
    else if (x < 0)
        return -1;
    else
        return 0;
}

double ErrorFunction(double x)
{
    double xSquared = x * x;

    double er = Sign(x) * sqrt(1 - exp(-xSquared * (4.0 / p_PI + a * xSquared) / (1 + a * xSquared)));

    return er;
}


double  DistributionFunction(double x, double mean, double variance)
{
    return 0.5 * (1 + ErrorFunction((x - mean) * 1.0 / (sqrt(variance) * s_SquareRoot2)));
}